src/NzbDrone.Core/Movies/MovieRepository.cs

```diff
@@ -32,195 +30,213 @@ public interface IMovieRepository : IBasicRepository<Movie>
+            {
+                movieEntry.AlternativeTitles.Add(altTitle);
+            }
+
+            return movieEntry;
         }
 
-        public Movie FindByTitle(string cleanTitle, int year)
+        protected override Movie SelectOne(SqlBuilder.Template sql)
         {
-            return FindByTitle(cleanTitle, year as int?);
+            return SelectMany(sql).FirstOrDefault();
         }
 
-        public Movie FindByImdbId(string imdbid)
+        protected override IEnumerable<Movie> SelectMany(SqlBuilder.Template sql)
         {
-            var imdbIdWithPrefix = Parser.Parser.NormalizeImdbId(imdbid);
-            return Query.Where(s => s.ImdbId == imdbIdWithPrefix).SingleOrDefault();
+            // Console.WriteLine($"Sql: {sql.RawSql}, Params: {((DynamicParameters)sql.Parameters).ToJsonString()}");
+            var movieDictionary = new Dictionary<int, Movie>();
+            return DataMapper.Query<Movie, Profile, AlternativeTitle, MovieFile, Movie>(
+                sql.RawSql,
+                (movie, profile, altTitle, file) => Map(movieDictionary, movie, profile, altTitle, file),
+                sql.Parameters)
+                .ToList();
         }
 
-        public List<Movie> GetMoviesByFileId(int fileId)
+        // This doesn't populate profiles - deserializing repeatedly is v slow
+        public override IEnumerable<Movie> All()
         {
-            return Query.Where(m => m.MovieFileId == fileId).ToList();
+            var noProfileTemplate = $"SELECT /**select**/ FROM {_table} /**leftjoin**/ /**where**/ /**orderby**/";
+            var sql = Builder().AddTemplate(noProfileTemplate);
+
+            var movieDictionary = new Dictionary<int, Movie>();
+            DataMapper.Query<Movie, AlternativeTitle, MovieFile, Movie>(
+                sql.RawSql,
+                (movie, altTitle, file) => Map(movieDictionary, movie, null, altTitle, file),
+                sql.Parameters);
+
+            // Doing the join on profile repo is slow - deserialize the same thing too many times
+            var profiles = _profileRepository.All();
+
+            return movieDictionary.Values.Join(profiles, m => m.ProfileId, p => p.Id, (movie, profile) => {
+                    movie.Profile = profile;
+                    return movie;
+                }).ToList();
         }
 
-        public void SetFileId(int fileId, int movieId)
+        public bool MoviePathExists(string path)
         {
-            SetFields(new Movie { Id = movieId, MovieFileId = fileId }, movie => movie.MovieFileId);
+            return Query(Builder().Where($"[{_table}].Path = @Path", new { Path = path })).Any();
         }
 
-        public Movie FindByTitleSlug(string slug)
+        public List<Movie> FindByTitles(List<string> titles)
         {
-            return Query.Where(m => m.TitleSlug == slug).FirstOrDefault();
+            return Query(Builder().Where($"[{_table}].CleanTitle IN @Titles OR AlternativeTitles.CleanTitle IN @Titles",
```

Comment:
No Profile and MovieFile here, going to be inconsistent.

Reply:
I think the Builder() should add in the join so you still get the extra stuff